Rules/AvoidTrailingWhitespace.cs

```diff
@@ -36,52 +36,67 @@ public IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
 
             var diagnosticRecords = new List<DiagnosticRecord>();
 
-            string[] lines = Regex.Split(ast.Extent.Text, @"\r?\n");
+            string[] lines = ast.Extent.Text.Split(new[] { "\r\n", "\r", "\n" }, StringSplitOptions.None);
```

Comment:
This makes me wonder: if we're just trying to find the extents of trailing whitespace, there's no need to split the string at all; we should just read through ourselves without allocating all these strings... But too much burden for this PR!

Reply:
Hmm, yh, I hear what you say, I guess for perf what counts is the 80-20 rule :-)
I'm aware of lot's of other small micro optimisations that one can make and even tried some but they didn't have a measurable outcome. Therefore I am focussed on just fixing what gives at least a measurable return.